solutions/gold/ioi-06-pyramid.mdx

```diff
@@ -0,0 +1,125 @@
+---
+id: ioi-06-pyramid
+source: IOI 2006
+title: Pyramid
+author: Justin Ji
+---
+
+## Explanation
+
+Consider iterating over every possible upper left corner of our pyramid's base.
+We want to find the chamber inside our pyramid's base with the smallest sum of
+heights.
+
+This boils down to doing a 2D range minimum query on every possible upper left
+corner of our chamber. The sliding window approach is as follows: when iterating
+over each row, we maintain an array of monotonic deques for
+each column that allows us to find the best chamber for each column. Then,
+we keep another monotonic deque that keeps track of the best possible result
+from our sliding window of columns.
```

Comment:
ngl explanation seems lazy

Reply:
ok that is pretty fair, but there isn't that much to this problem other than writing the impl

Reply:
oh yeah about this uhh do u have any tips for making the explanation more thought out, per see

cuz I was lowk blanking out on how I could do more to explain the sol

Reply:
i mean an array of monotonic deques
what is it indexed by?
what does each deque contain?
how/when do we add/pop from it?

Reply:
ok i went a bit more in depth with the explanation. lmk if any part of the explanation is unclear